### ISSUE TITLE
'NoneType' object has no attribute '__getitem__'

### ISSUE THREAD

**Issue by mpizosdim** [llSourcell/How_to_make_a_text_summarizer#9]
@llSourcell @rtlee9  The following error come up in `'NoneType' object has no attribute '__getitem__'`:

`activation_energies = activation_energies + -1e20*K.expand_dims(1.-K.cast(mask[:, :maxlend],'float32'),1)`. 

This is happening when running:
`model.add(SimpleContext(name='simplecontext_1'))`

Using:

`keras.__version__: 2.0.4`
`tf.__version__:  1.3.0`

Any idea why is this happening?
I think is a good idea to add a DockerFile in your projects. thnks

**Comment by sanniheruwala** [comment]
Hey i am also getting same error ... please share your solution if you have found it

**Comment by mzhao98** [comment]
Hi, I'm also getting the same error when running SimpleContext. I know it's late, but has anyone solved this issue?

**Comment by nalin-mathur** [comment]
I am getting the same error when running my script on python2.7. But when I run the same on python3.6 I get this error: 'NoneType' object is not subscriptable .

The problem with the code is that the object 'mask' is a NoneType object as it does not hold any default value whereas the other variables in the function do. Hence when it is called from within the SimpleContext class from the line 'super(SimpleContext, self).__init__(simple_context,**kwargs)', no argument is passed to the mask object. 

Hence it is a NoneType object which is generating the error. 
I am not sure what is being done by the code, otherwise I would've tried to find a way around it.

**Comment by ghost** [comment]
Answer is that you use Keras 2 and interface for custom Layer or Lambda layer change

instead of 

> get_output_shape_for 

use 

> compute_output_shape

**Comment by vighnesh153** [comment]
Works like a charm.
